Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -0,0 +1,957 @@
+codeunit 40900 "GP Populate Hist. Tables"
+{
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        HistMigrationStatusMgmt: Codeunit "Hist. Migration Status Mgmt.";
+        PayrollSeriesId: Integer;
+
+    trigger OnRun()
+    begin
+        PayrollSeriesId := 6;
+        PopulateHistoricalTables();
+    end;
+
+    internal procedure PopulateHistoricalTables()
+    begin
+        if GPCompanyAdditionalSettings.GetMigrateHistory() then begin
+            PopulateGLDetail();
+            PopulateReceivables();
+            PopulatePayablesDocuments();
+            PopulateInventoryTransactions();
+            PopulatePurchaseRecvTransactions();
+            HistMigrationStatusMgmt.SetStatusFinished();
+        end;
+    end;
+
+    local procedure PopulateGLDetail()
+    begin
+        // GP G/L Accounts
+        if HistMigrationStatusMgmt.HasNotRanStep("Hist. Migration Step Type"::"GP GL Accounts") then
+            if GPCompanyAdditionalSettings.GetMigrateHistGLTrx() then begin
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Accounts", false);
+                PopulateHistGLAccounts();
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Accounts", true);
+            end;
+
+        // GP G/L Journal Trx.
+        if HistMigrationStatusMgmt.HasNotRanStep("Hist. Migration Step Type"::"GP GL Journal Trx.") then
+            if GPCompanyAdditionalSettings.GetMigrateHistGLTrx() then begin
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", false);
+                PopulateOpenYearPostedTransactions();
+                PopulateHistoricalYearPostedTransactions();
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", true);
+            end;
+    end;
+
+    local procedure PopulateHistGLAccounts()
+    var
+        GPGL00105: Record "GP GL00105";
+        GPGL00100: Record "GP GL00100";
+        HistGLAccount: Record "Hist. G/L Account";
+    begin
+        if GPGL00105.FindSet() then
+            repeat
+                Clear(HistGLAccount);
+                if GPGL00100.Get(GPGL00105.ACTINDX) then begin
+                    HistGLAccount."No." := CopyStr(Format(GPGL00105.ACTNUMST), 1, MaxStrLen(HistGLAccount."No."));
+                    HistGLAccount.Name := GPGL00100.ACTDESCR;
+
+                    if not HistGLAccount.Insert() then
+                        HistMigrationStatusMgmt.ReportError("Hist. Migration Step Type"::"GP GL Accounts", GPGL00105.ACTNUMST,
+                            GetLastErrorCode(), GetLastErrorText());
```

Comment:
If we follow this approach we will need to track which records failed (maybe within the error record), thus we can migrate the failed records after the job. 

As an alternative we would need to stop the job (if we cannot update the records in isolation). First approach is better.

Reply:
I have added batched deletes in the background job, and the batch deletes are attempted when re-ran. This is the simpler approach, which I prefer right now. If we are able to move toward DataTransfer, then the 'if not table.Insert()' pattern won't be used anymore.